heft-plugins/heft-storybook-plugin/src/StorybookPlugin.ts

```diff
@@ -0,0 +1,237 @@
+            // Discard Webpack's configuration to prevent Webpack from running
+            return null;
+          }
+        );
+
+        bundle.hooks.run.tapPromise(PLUGIN_NAME, async () => {
+          await this._onBundleRunAsync(heftSession, heftConfiguration);
+        });
+      });
+    });
+  }
+
+  private _onPreCompile(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    this._logger.terminal.writeVerboseLine(`Probing for "${this._storykitPackageName}"`);
+
+    // Example: "/path/to/my-project/node_modules/my-storykit"
+    let storykitFolder: string;
+    try {
+      storykitFolder = Import.resolvePackage({
+        packageName: this._storykitPackageName,
+        baseFolderPath: heftConfiguration.buildFolder
+      });
+    } catch (ex) {
+      throw new Error(`The ${TASK_NAME} task cannot start: ` + ex.message);
+    }
+
+    this._logger.terminal.writeVerboseLine(`Found "${this._storykitPackageName}" in ` + storykitFolder);
+
+    // Example: "/path/to/my-project/node_modules/my-storykit/node_modules"
+    const storykitModuleFolder: string = path.join(storykitFolder, 'node_modules');
+    if (!FileSystem.exists(storykitModuleFolder)) {
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because the storykit module folder does not exist:\n` +
+          storykitModuleFolder +
+          '\nDid you forget to install it?'
+      );
+    }
+
+    this._logger.terminal.writeVerboseLine(`Resolving startupModulePath "${this._startupModulePath}"`);
+    try {
+      this._resolvedStartupModulePath = Import.resolveModule({
+        modulePath: this._startupModulePath,
+        baseFolderPath: storykitModuleFolder
+      });
+    } catch (ex) {
+      throw new Error(`The ${TASK_NAME} task cannot start: ` + ex.message);
+    }
+    this._logger.terminal.writeVerboseLine(
+      `Resolved startupModulePath is "${this._resolvedStartupModulePath}"`
+    );
+
+    // Example: "/path/to/my-project/.storybook"
+    const dotStorybookFolder: string = path.join(heftConfiguration.buildFolder, '.storybook');
+    FileSystem.ensureFolder(dotStorybookFolder);
+
+    // Example: "/path/to/my-project/.storybook/node_modules"
+    const dotStorybookModuleFolder: string = path.join(dotStorybookFolder, 'node_modules');
+
+    // Example:
+    //   LINK FROM: "/path/to/my-project/.storybook/node_modules"
+    //   TARGET:    "/path/to/my-project/node_modules/my-storykit/node_modules"
+    this._createFolderSymlink({
+      newLinkPath: dotStorybookModuleFolder,
+      linkTargetPath: storykitModuleFolder
+    });
+  }
+
+  private _createFolderSymlink(options: IFileSystemCreateLinkOptions): void {
+    // TODO: createSymbolicLinkJunction() is supposed to implement alreadyExistsBehavior
+    FileSystem.deleteFile(options.newLinkPath);
+
+    // This is how Rush links node_modules folders, for some historical reasons
+    // related to filesystem permissions
+    if (process.platform === 'win32') {
```

Comment:
Should we just expose an API in `FileSystem` for this?

Reply:
It may be the case that `createSymbolicLinkJunction()` already embodies these semantics. (?) I didn't have time to test it.

Reply:
@iclanton I researched this further.  The best practices are difficult to explain, so I think we should wrap it up in a helper API like `createSmartLinkFile()` / `createSmartLinkFolder()`, but I will tackle that in a separate PR.  I did write up the guidance as API docs in this PR, though.